content/topics/java-specific/oop-basics/_index.md

```diff
@@ -1,15 +1,179 @@
 ---
 title: Java OOP basics resources and readings
+ready: true
 ---
 
-TODO:
+## OOP  in Java
+The main aim of OOP is to bind together the data and the functions that operate on them.This is done so that no other part of the code can access this data except that function.
 
-- class
-- calling functions
-- instantiation of classes
+There are four OOP priciples in java:
 
-- encapsulation
-- inheritence
-- etc...
+### Abstraction
+In Object-oriented programming, abstraction is a process of hiding the implementation details from the user, only the functionality will be provided to the user. In other words, the user will have the information on what the object does instead of how it does it.
 
-Either find a nice resuorce or demonstrate with a nice example. Eg: Farm contains Animals and Plants, Cow is an Animal, Potatoe is a plant etc.
+In Java, abstraction is achieved using Abstract classes and interfaces.[Read more](https://www.geeksforgeeks.org/abstraction-in-java-2/)
+
+The following Java code shows how Abstraction can be implemented:
+
+```
+//abstract parent class
+abstract class Animal{
+    //abstract method
+    public abstract void sound();
+    //regular method
+    public void sleep() {
+    System.out.println("Zzz");
+  }
+}
+//Dog class extends Animal class
+class Dog extends Animal{
+
+    public void sound(){
+        System.out.println("Woof");
+    }
+    public static void main(String args[]){
+        Animal obj = new Dog();
+        obj.sound();
+    }
+}
+ 
+``` 
+
+
+### Encapsulation
+Encapsulation is known as data-hiding.
+It is an OOP principle which states that all the attributes and behaviours (methods and functions) of an object should be grouped together in one datatype or class.
+This means that objects may be able to communicate with one another but are restricted to access some of the object's components directly.
```

Comment:
Remember that we are dealing with noobs here. Your wording has to be ultra clear because they will lack the context to decipher your meaning.

Reply:
Added a link that provides simpler explanation